fetch/api/request/request-consume-empty.html

```diff
@@ -61,20 +61,32 @@
      });
     }
 
-    function checkRequestWithNoBody(bodyType, checkFunction) {
+    function checkBodyFormDataError(request) {
+      return request.formData().then(
+        function(bodyAsFormData) {
+          assert_unreached("FormData parsing should fail");
+        },
+        function(e) {
+          assert_throws(new TypeError(), function() { throw e; });
+          assert_false(request.bodyUsed);
+        });
```

Comment:
It seems you could replace most of this with `return promise_reject(new TypeError(), request.formData())` or something close to that, no?